packages/hooks/src/useCurrentElement.ts

```diff
@@ -16,13 +16,12 @@ export const findMountedDomRecursive = <T extends Element>(
 }
 
 export const useCurrentElement = <T extends Element>() => {
-  if (!sideEffectsEnabled()) return
-
-  const [elm, setElm] = useState<T | undefined>(undefined)
+  const elm = signal<T | undefined>(undefined)
+  if (!sideEffectsEnabled()) return elm
 
   return useHook("useCurrentElement", {}, ({ vNode, queueEffect }) => {
     queueEffect(() => {
-      setElm(findMountedDomRecursive<T>(vNode.child))
+      elm.value = findMountedDomRecursive<T>(vNode.child)
```

Comment:
we should do an identity check here. that would allow us not to have to do `.peak` in useCurrentElement

something like this
```ts
const newElm = findMountedDomRecursive<T>(vNode.child)
if (newElm != elm.value) {
  elm.value = newElm
```